src/cpp/rtps/builtin/discovery/participant/PDP.cpp

```diff
@@ -1698,6 +1700,123 @@ void PDP::add_builtin_security_attributes(
 
 #endif // HAVE_SECURITY
 
+void PDP::local_participant_attributes_update_nts(
+        const RTPSParticipantAttributes& new_atts)
+{
+    // Update user data
+    auto participant_data = getLocalParticipantProxyData();
+    participant_data->m_userData.data_vec(new_atts.userData);
+
+    bool announce_locators = !mp_RTPSParticipant->is_intraprocess_only();
+    if (!announce_locators)
+    {
+        // If we are intraprocess only, we do not need to update locators
+        return;
+    }
```

Comment:
Why don't we simplify this?
```suggestion
    if (mp_RTPSParticipant->is_intraprocess_only())
    {
        // If we are intraprocess only, we do not need to update locators
        return;
    }
```

Reply:
I like that, yes